roles/flink/tasks/main.yml

```diff
@@ -38,7 +39,12 @@
     https_proxy: "{{http_proxy_url}}"
 
 - name: Unpack tarball
-  command: tar xzf {{flink_dir}}/flink-{{flink_version}}.tgz --strip-components=1 chdir={{flink_dir}} creates={{flink_dir}}/bin
+  unarchive:
+    src: "{{flink_dir}}/flink-{{flink_version}}.tgz"
+    dest: "{{flink_dir}}"
+    extra_opts:
+    - --strip-components=1
+    remote_src: yes
```

Comment:
Previously the final structure of the files:

```bash
[root@flink1 ~]# tree /opt/flink_1.3.2/
/opt/flink_1.3.2/
├── bin
│   ├── config.sh
│   ├── flink
│   ├── flink.bat
│   ├── flink-console.sh
│   ├── flink-daemon.sh
│   ├── historyserver.sh
│   ├── jobmanager.sh
│   ├── mesos-appmaster.sh
│   ├── mesos-taskmanager.sh
│   ├── pyflink.bat
│   ├── pyflink.sh
│   ├── start-cluster.sh
│   ├── start-local.bat
│   ├── start-local.sh
│   ├── start-scala-shell.sh
│   ├── start-zookeeper-quorum.sh
│   ├── stop-cluster.sh
│   ├── stop-local.sh
│   ├── stop-zookeeper-quorum.sh
│   ├── taskmanager.sh
│   ├── yarn-session.sh
│   └── zookeeper.sh
├── conf
│   ├── flink-conf.yaml
│   ├── log4j-cli.properties
│   ├── log4j-console.properties
│   ├── log4j.properties
│   ├── log4j-yarn-session.properties
│   ├── logback-console.xml
│   ├── logback.xml
│   ├── logback-yarn.xml
│   ├── masters
│   ├── slaves
│   └── zoo.cfg
├── examples
│   ├── batch
│   │   ├── ConnectedComponents.jar
│   │   ├── DistCp.jar
│   │   ├── EnumTriangles.jar
│   │   ├── KMeans.jar
│   │   ├── PageRank.jar
│   │   ├── TransitiveClosure.jar
│   │   ├── WebLogAnalysis.jar
│   │   └── WordCount.jar
│   ├── gelly
│   │   └── flink-gelly-examples_2.10-1.3.2.jar
│   ├── python
│   │   ├── __init__.py
│   │   ├── TPCHQuery10.py
│   │   ├── TPCHQuery3.py
│   │   ├── TriangleEnumeration.py
│   │   ├── WebLogAnalysis.py
│   │   └── WordCount.py
│   └── streaming
│       ├── IncrementalLearning.jar
│       ├── Iteration.jar
│       ├── Kafka.jar
│       ├── SessionWindowing.jar
│       ├── SocketWindowWordCount.jar
│       ├── TopSpeedWindowing.jar
│       ├── Twitter.jar
│       ├── WindowJoin.jar
│       └── WordCount.jar
├── flink-1.3.2.tgz
├── lib
│   ├── flink-dist_2.10-1.3.2.jar
│   ├── flink-python_2.10-1.3.2.jar
│   ├── flink-shaded-hadoop2-uber-1.3.2.jar
│   ├── log4j-1.2.17.jar
│   └── slf4j-log4j12-1.7.7.jar
├── LICENSE
├── log
│   ├── flink-root-jobmanager-0-flink1.priv.log
│   ├── flink-root-jobmanager-0-flink1.priv.out
│   ├── flink-root-taskmanager-0-flink1.priv.log
│   └── flink-root-taskmanager-0-flink1.priv.out
├── NOTICE
├── opt
│   ├── flink-cep_2.10-1.3.2.jar
│   ├── flink-cep-scala_2.10-1.3.2.jar
│   ├── flink-gelly_2.10-1.3.2.jar
│   ├── flink-gelly-scala_2.10-1.3.2.jar
│   ├── flink-metrics-datadog-1.3.2.jar
│   ├── flink-metrics-dropwizard-1.3.2.jar
│   ├── flink-metrics-ganglia-1.3.2.jar
│   ├── flink-metrics-graphite-1.3.2.jar
│   ├── flink-metrics-statsd-1.3.2.jar
│   ├── flink-ml_2.10-1.3.2.jar
│   └── flink-table_2.10-1.3.2.jar
├── README.txt
├── resources
│   └── python
│       ├── flink
│       │   ├── connection
│       │   │   ├── Collector.py
│       │   │   ├── Connection.py
│       │   │   ├── Constants.py
│       │   │   ├── __init__.py
│       │   │   └── Iterator.py
│       │   ├── functions
│       │   │   ├── Aggregation.py
│       │   │   ├── CoGroupFunction.py
│       │   │   ├── CrossFunction.py
│       │   │   ├── FilterFunction.py
│       │   │   ├── FlatMapFunction.py
│       │   │   ├── Function.py
│       │   │   ├── GroupReduceFunction.py
│       │   │   ├── __init__.py
│       │   │   ├── JoinFunction.py
│       │   │   ├── KeySelectorFunction.py
│       │   │   ├── MapFunction.py
│       │   │   ├── MapPartitionFunction.py
│       │   │   ├── ReduceFunction.py
│       │   │   └── RuntimeContext.py
│       │   ├── __init__.py
│       │   ├── plan
│       │   │   ├── Constants.py
│       │   │   ├── DataSet.py
│       │   │   ├── Environment.py
│       │   │   ├── __init__.py
│       │   │   └── OperationInfo.py
│       │   └── utilities
│       │       └── __init__.py
│       ├── __init__.py
│       └── setup.py
└── tools
    └── planVisualizer.html

18 directories, 110 files
```
```bash
[root@hdfs2 ~]# ls -alh /opt/
total 16K
drwxr-xr-x.  4 root         root         4.0K Apr  9 14:35 .
dr-xr-xr-x. 19 root         root         4.0K Apr  9 16:30 ..
drwxr-xr-x  10 flink        flink        4.0K Apr  9 14:36 flink_1.3.2
```


```bash
[root@hdfs2 ~]# ls -alh /opt/flink_1.3.2/
total 136M
drwxr-xr-x  10 flink flink    4.0K Apr  9 14:36 .
drwxr-xr-x.  4 root  root     4.0K Apr  9 14:35 ..
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 bin
drwxrwxr-x   2  1005 eimamagi 4.0K Apr  9 14:36 conf
drwxrwxr-x   6  1005 eimamagi 4.0K Aug  3  2017 examples
-rw-r--r--   1 root  root     136M Apr  9 14:35 flink-1.3.2.tgz
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 lib
-rw-r--r--   1  1005 eimamagi  18K Jul 28  2017 LICENSE
drwxrwxr-x   2  1005 eimamagi 4.0K Apr  9 16:51 log
-rw-r--r--   1  1005 eimamagi  779 Jul 28  2017 NOTICE
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 opt
-rw-r--r--   1  1005 eimamagi 1.3K Jul 28  2017 README.txt
drwxrwxr-x   3  1005 eimamagi 4.0K Aug  3  2017 resources
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 tools
```

-------------------------------------------------------------------------

Now they also have the same structure:

```bash
[root@flink1 ~]# tree /opt/flink_1.3.2/
/opt/flink_1.3.2/
├── bin
│   ├── config.sh
│   ├── flink
│   ├── flink.bat
│   ├── flink-console.sh
│   ├── flink-daemon.sh
│   ├── historyserver.sh
│   ├── jobmanager.sh
│   ├── mesos-appmaster.sh
│   ├── mesos-taskmanager.sh
│   ├── pyflink.bat
│   ├── pyflink.sh
│   ├── start-cluster.sh
│   ├── start-local.bat
│   ├── start-local.sh
│   ├── start-scala-shell.sh
│   ├── start-zookeeper-quorum.sh
│   ├── stop-cluster.sh
│   ├── stop-local.sh
│   ├── stop-zookeeper-quorum.sh
│   ├── taskmanager.sh
│   ├── yarn-session.sh
│   └── zookeeper.sh
├── conf
│   ├── flink-conf.yaml
│   ├── log4j-cli.properties
│   ├── log4j-console.properties
│   ├── log4j.properties
│   ├── log4j-yarn-session.properties
│   ├── logback-console.xml
│   ├── logback.xml
│   ├── logback-yarn.xml
│   ├── masters
│   ├── slaves
│   └── zoo.cfg
├── examples
│   ├── batch
│   │   ├── ConnectedComponents.jar
│   │   ├── DistCp.jar
│   │   ├── EnumTriangles.jar
│   │   ├── KMeans.jar
│   │   ├── PageRank.jar
│   │   ├── TransitiveClosure.jar
│   │   ├── WebLogAnalysis.jar
│   │   └── WordCount.jar
│   ├── gelly
│   │   └── flink-gelly-examples_2.10-1.3.2.jar
│   ├── python
│   │   ├── __init__.py
│   │   ├── TPCHQuery10.py
│   │   ├── TPCHQuery3.py
│   │   ├── TriangleEnumeration.py
│   │   ├── WebLogAnalysis.py
│   │   └── WordCount.py
│   └── streaming
│       ├── IncrementalLearning.jar
│       ├── Iteration.jar
│       ├── Kafka.jar
│       ├── SessionWindowing.jar
│       ├── SocketWindowWordCount.jar
│       ├── TopSpeedWindowing.jar
│       ├── Twitter.jar
│       ├── WindowJoin.jar
│       └── WordCount.jar
├── flink-1.3.2.tgz
├── lib
│   ├── flink-dist_2.10-1.3.2.jar
│   ├── flink-python_2.10-1.3.2.jar
│   ├── flink-shaded-hadoop2-uber-1.3.2.jar
│   ├── log4j-1.2.17.jar
│   └── slf4j-log4j12-1.7.7.jar
├── LICENSE
├── log
│   ├── flink-root-jobmanager-0-flink1.priv.log
│   ├── flink-root-jobmanager-0-flink1.priv.out
│   ├── flink-root-taskmanager-0-flink1.priv.log
│   └── flink-root-taskmanager-0-flink1.priv.out
├── NOTICE
├── opt
│   ├── flink-cep_2.10-1.3.2.jar
│   ├── flink-cep-scala_2.10-1.3.2.jar
│   ├── flink-gelly_2.10-1.3.2.jar
│   ├── flink-gelly-scala_2.10-1.3.2.jar
│   ├── flink-metrics-datadog-1.3.2.jar
│   ├── flink-metrics-dropwizard-1.3.2.jar
│   ├── flink-metrics-ganglia-1.3.2.jar
│   ├── flink-metrics-graphite-1.3.2.jar
│   ├── flink-metrics-statsd-1.3.2.jar
│   ├── flink-ml_2.10-1.3.2.jar
│   └── flink-table_2.10-1.3.2.jar
├── README.txt
├── resources
│   └── python
│       ├── flink
│       │   ├── connection
│       │   │   ├── Collector.py
│       │   │   ├── Connection.py
│       │   │   ├── Constants.py
│       │   │   ├── __init__.py
│       │   │   └── Iterator.py
│       │   ├── functions
│       │   │   ├── Aggregation.py
│       │   │   ├── CoGroupFunction.py
│       │   │   ├── CrossFunction.py
│       │   │   ├── FilterFunction.py
│       │   │   ├── FlatMapFunction.py
│       │   │   ├── Function.py
│       │   │   ├── GroupReduceFunction.py
│       │   │   ├── __init__.py
│       │   │   ├── JoinFunction.py
│       │   │   ├── KeySelectorFunction.py
│       │   │   ├── MapFunction.py
│       │   │   ├── MapPartitionFunction.py
│       │   │   ├── ReduceFunction.py
│       │   │   └── RuntimeContext.py
│       │   ├── __init__.py
│       │   ├── plan
│       │   │   ├── Constants.py
│       │   │   ├── DataSet.py
│       │   │   ├── Environment.py
│       │   │   ├── __init__.py
│       │   │   └── OperationInfo.py
│       │   └── utilities
│       │       └── __init__.py
│       ├── __init__.py
│       └── setup.py
└── tools
    └── planVisualizer.html

18 directories, 110 files
```

```bash
[root@flink1 ~]# ls -alh /opt/
total 12K
drwxr-xr-x.  3 root  root  4.0K Apr 10 10:23 .
dr-xr-xr-x. 18 root  root  4.0K Apr 10 08:23 ..
drwxr-xr-x  10 flink flink 4.0K Apr 10 11:47 flink_1.3.2
```


```bash
[root@flink1 ~]# ls -alh /opt/flink_1.3.2/
total 136M
drwxr-xr-x  10 flink flink    4.0K Apr 10 11:47 .
drwxr-xr-x.  3 root  root     4.0K Apr 10 10:23 ..
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 bin
drwxrwxr-x   2  1005 eimamagi 4.0K Apr 10 11:47 conf
drwxrwxr-x   6  1005 eimamagi 4.0K Aug  3  2017 examples
-rw-r--r--   1 root  root     136M Apr 10 11:47 flink-1.3.2.tgz
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 lib
-rw-r--r--   1  1005 eimamagi  18K Jul 28  2017 LICENSE
drwxrwxr-x   2  1005 eimamagi 4.0K Jul 28  2017 log
-rw-r--r--   1  1005 eimamagi  779 Jul 28  2017 NOTICE
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 opt
-rw-r--r--   1  1005 eimamagi 1.3K Jul 28  2017 README.txt
drwxrwxr-x   3  1005 eimamagi 4.0K Aug  3  2017 resources
drwxrwxr-x   2  1005 eimamagi 4.0K Aug  3  2017 tools
```